File: js/2_EvenFib.js
function main(limit) {
	fib = [0, 1];
	len = 2; 
	sum = 0; 
	val = 1;
	
	while (val < limit) {
		val = fib[len-1] + fib[len-2];
		fib.push(fib[len-1]+fib[len-2]);
		if (val % 2 == 0) { sum += val; }
		len += 1;
	}
	return(sum);
}

console.log(main(4000000));

File: haskell/2_EvenFib.hs
main = putStrLn (show val)
val = sum $ filter even $ takeWhile (<= 4000000) [fib x | x <- [1..]]

fib 1 = 1
fib 2 = 2
fib x = fib (x - 1) + fib (x - 2)